src/test/java/org/opensearch/knn/bwc/KNNBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,193 @@
+import org.opensearch.knn.KNNResult;
+import org.opensearch.knn.index.KNNQueryBuilder;
+import org.opensearch.knn.KNNRestTestCase;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.junit.Assert;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+public class KNNBackwardsCompatibilityIT extends KNNRestTestCase {
+    private static final ClusterType CLUSTER_TYPE = ClusterType.parse(System.getProperty("tests.rest.bwcsuite"));
+    private static final String CLUSTER_NAME = System.getProperty("tests.clustername");
+    private final String testIndexName = "knn-bwc-test-index";
+    private final String testFieldName = "knn-bwc-test-field";
+    private final int dimensions = 2;
+
+    @Override
+    protected final boolean preserveIndicesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final boolean preserveReposUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveTemplatesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final Settings restClientSettings() {
+        return Settings
+                .builder()
+                .put(super.restClientSettings())
+                // increase the timeout here to 90 seconds to handle long waits for a green
+                // cluster health. the waits for green need to be longer than a minute to
+                // account for delayed shards
+                .put(OpenSearchRestTestCase.CLIENT_SOCKET_TIMEOUT, "90s")
+                .build();
+    }
+
+    private enum ClusterType {
+        OLD,
+        MIXED,
+        UPGRADED;
+
+        public static ClusterType parse(String value) {
+            switch (value) {
+                case "old_cluster":
+                    return OLD;
+                case "mixed_cluster":
+                    return MIXED;
+                case "upgraded_cluster":
+                    return UPGRADED;
+                default:
+                    throw new AssertionError("unknown cluster type: " + value);
+            }
+        }
+    }
+
+// Use this prefix "knn-bwc-" while creating a test index to test BWC Tests.
+// For example:  testIndexName = "knn-bwc-test-index"
+    @SuppressWarnings("unchecked")
+    public void testBackwardsCompatibility() throws Exception {
+        String uri = getUri();
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<Object> pluginNames = plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
+            switch (CLUSTER_TYPE) {
+                case OLD:
+                    Assert.assertTrue(pluginNames.contains("opensearch-knn"));
```

Comment:
we should have constant for plugin name if not, lets create at least in this test class for now.

Reply:
Done